src/vision/VisionProcessor.cpp

```diff
@@ -0,0 +1,91 @@
+#include "VisionProcessor.hpp"
+
+//paramGroup defines the subdirectory under /vision to look for and use in params
+VisionProcessor::VisionProcessor(string paramGroup)
+{
+    this->paramGroup = paramGroup;
+}
+
+VisionProcessor::~VisionProcessor()
+{
+
+}
+
+//Processes the image using color filtering with parameters under given subgroup
+Mat VisionProcessor::process(Image& image)
+{
+    //Create a nodehandle for param fetching
+    ros::NodeHandle n;
+    
+    //Convert image message to OpenCV Mat
+    Mat toProcess = toOpenCV(image);
+    
+    ROS_DEBUG_STREAM("Image now OpenCv Mat");
+
+    //Convert to HSV color space
+    Mat hsv;
+    cvtColor(toProcess, hsv, cv::COLOR_BGR2HSV);
+
+    ROS_DEBUG_STREAM("Converted image to hsv");
+
+    //Create the lowerbound of thresholding with defaults
+    Scalar lower_bound(0, 10, 10);
+    
+    //Get lowerbound params
+    n.getParamCached("/vision/" + paramGroup + "/min/hue", lower_bound[0]);
+    n.getParamCached("/vision/" + paramGroup + "/min/sat", lower_bound[1]);
+    n.getParamCached("/vision/" + paramGroup + "/min/val", lower_bound[2]);
+
+    //Create upperbound of thresholding with defaults
+    Scalar upper_bound(10,255,255);
+ 
+    //Get upperbound params
+    n.getParamCached("/vision/" + paramGroup + "/max/hue", upper_bound[0]);
+    n.getParamCached("/vision/" + paramGroup + "/max/sat", upper_bound[1]);
+    n.getParamCached("/vision/" + paramGroup + "/max/val", upper_bound[2]);   
+
+    ROS_DEBUG_STREAM("Performing Masking and/or other processing");
+    Mat mask;
+    //Blur the image for a bit more smoothness
+    medianBlur(mask, mask, 3);
```

Comment:
Will fix. I had the blur after the inRange, but then moved it and forgot to change this.